### DISCUSSION TITLE
Possible typo in TELEGRAM_CHAT_ID?

### DISCUSSION THREAD

**Discussion by jackblk** [nikhilbadyal/docker-py-revanced#441] [Q&A]
The traceback threw `INPUT_CHAT_ID` instead of `TELEGRAM_CHAT_ID`.

I trace a bit and I found that the names are different, but should be mapped correctly.

* Workflow: https://github.com/nikhilbadyal/ghactions/blob/main/.github/workflows/telegram-uploader.yml#L37
* Uploader workflow code: https://github.com/nikhilbadyal/ghaction-telegram-uploader/blob/main/src/config.py#L21C35

Here's the log

```
Traceback (most recent call last):
  File "/app/main.py", line 23, in <module>
    loop.run_until_complete(main())
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/app/main.py", line 15, in main
    config = UploaderConfig(env)
             ^^^^^^^^^^^^^^^^^^^
  File "/app/src/config.py", line 21, in __init__
    self.chat_id = env.int("INPUT_CHAT_ID")
                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/environs/__init__.py", line 126, in method
    raise EnvValidationError(
environs.EnvValidationError: Environment variable "INPUT_CHAT_ID" invalid: Not a valid integer.
```

Am I doing something wrong?

**Comment by nikhilbadyal** [comment]
It's fine actually. 
The base workflow expects [`CHAT_ID`](https://github.com/nikhilbadyal/ghaction-telegram-uploader/blob/c43513751d3f6723c19b66b141eb48617690794f/action.yml#L14) as input. 
Docker Py Revanced accepts [`TELEGRAM_CHAT_ID`](https://github.com/nikhilbadyal/docker-py-revanced#telegram-support) to keep things sane and it passes the value to the middle workflow as [`TELEGRAM_CHAT_ID`](https://github.com/nikhilbadyal/docker-py-revanced/blob/8d6f942b682190dbf075e413064ce9ae12c9868e/.github/workflows/build-apk.yml#L119) and the middle workflow then [passes](https://github.com/nikhilbadyal/ghactions/blob/609dc5580636f4e655a8ad5bf53f31ca1612a098/.github/workflows/telegram-uploader.yml#L37) `TELEGRAM_CHAT_ID`  as `CHAT_ID` to the base workflow.


Now coming to the `INPUT_` - 

The piece of code expect [`INPUT_CHAT_ID`](https://github.com/nikhilbadyal/ghaction-telegram-uploader/blob/c43513751d3f6723c19b66b141eb48617690794f/src/config.py#L21C35) instead of `CHAT_ID` in base workflow. As Github automatically add `INPUT_` [in front ](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs) of the all the [envs](https://stackoverflow.com/a/58243967/12136319).

Now coming to the error message - 

The error message also validate above information as it states the type of `CHAT_ID` is not integer and it doesn't complain that the variable is missing. So the base workflow actually received the input but the input was not correct. CHAT_ID is supposed to be a integer(numerical value like 1,2,3) value.


I think the `TELEGRAM_CHAT_ID` you provided is incorrect. Mind sharing it here so I can take a look ?

**Reply by jackblk** [reply]
I see. Should `TELEGRAM_CHAT_ID` include the minus symbol (`-1000000`) because it's a chat channel?

**Reply by nikhilbadyal** [reply]
Send your channel name to this [bot](https://t.me/raw_data_bot).
 It can give you the correct CHAT_ID.

**Reply by jackblk** [reply]
Hey, thanks for the answer. I think I found the issue with it.

I naively bundled all envs `TELEGRAM_*` in 1 secret `ENVS`, but `ENVS` was meant only for 1 workflow, it is unpacked in only "build apk" workflow. I added them separately in repo secrets and it's working now.

Thanks again for a great repo, it works well!
